### ISSUE TITLE
Website: Code samples missing when JavaScript disabled.

### ISSUE THREAD

**Issue by juliaelman** [18F/web-design-standards#1305]
## Bug

* Website
* Accessibility

## Description

The WDS UI Components come with code samples for users to reference as needed. When JavaScript is disabled the code sample boxes no longer appear. See steps to reproduce below for reference.

Referencing #1224 for debugging purposes.

## Steps to reproduce the issue

1. Disable JavaScript in your browser
2. Go to `localhost:4000/accordions`
3. Scroll down to area below interactive example

![pe1c6dd6od](https://cloud.githubusercontent.com/assets/955558/16658358/13df42c4-4434-11e6-8371-1747b94f7c03.gif)

**Comment by juliaelman** [comment]
This issue is based off of the premise that code samples are [generated with JS](https://github.com/18F/web-design-standards/blob/staging/docs/doc_assets/js/styleguide.js#L20). The team is determining whether or not a Jekyll plugin makes sense as the best path forward.

**Comment by rogeruiz** [comment]
As @juliaelman mentioned above, the root cause is that the code snippets aren't being generated at build time. Right now the code snippets live in partials that are not being consumed more than once during the build step for the site. They are being rendered on the page above the accordion. We need another partial which takes the code and places it a block within an accordion. Which is all the JS is doing but on the client instead of during the build. 

By moving the code snippets generation to the server, we can also remove the Prisim dependency on the front end and just use the version of Prisim that Jekyll provides.

**Comment by maya** [comment]
>and just use the version of Prisim that Jekyll provides
To my knowledge, Jekyll doesn't use Prism, it uses Pygments. Prism is superior for syntax highlighting.

**Comment by rogeruiz** [comment]
@maya yes. Sorry about that. We will need to use something that Jekyll supports. That also means that the styles for the code blocks will need to be updated.

There's another option here, and that is using gulp to convert the code blocks using Prisim and then giving that data to Jekyll. It might be worth doing if the styling of the code blocks is going to be a big effort.

**Comment by joshbruce** [comment]
Just to throw it out there as a different perspective/curiosity:

Is the automation of, what essentially a copy-paste operation, necessary? Can't contributors just copy and paste the preview code into a code block `div`? How often are the previews changed or created?

Again, not offering an opinion on either side here, just some curiosities toward an alternative perspective.

**Comment by rogeruiz** [comment]
@joshbruce great question. The automation is essentially a copy-paste operation that needs to occur on the build step of the website. Currently, code blocks are injected via JS on all the pages that render them and the issue is that without JS there are no code blocks. Automating this would allow for the build step to do this processing and rendering on the server before the site is built and deployed. That way we can [remove the dependency on this functionality](https://github.com/18F/web-design-standards/blob/staging/docs/doc_assets/js/components/generate-code-snippets.js) on the client side. 

Also automating this would mean that as contributors and maintainers, we would only need to update a single version of the code blocks instead of maintaining multiple `HTML` copies of the components in separate files.

**Comment by maya** [comment]
Fixed in #1370.
